vm-setup/roles/v1aX_integration_test/tasks/node_reuse.yml

```diff
@@ -159,12 +159,25 @@
       - /tmp/before_upgrade_mapping.txt
       - /tmp/after_upgrade_mapping.txt
 
+  - name: Get resourceVersion of KCP
+    shell: |
+        kubectl get kubeadmcontrolplane "{{ CLUSTER_NAME }}" -n "{{ NAMESPACE }}" -o json | jq '.metadata.resourceVersion'
+    register: resource_version_kcp 
+    environment:
+      KUBECONFIG: "/tmp/kubeconfig-{{ CLUSTER_NAME }}.yaml"
+
   - name: Put maxSurge field in KubeadmControlPlane back to default value(1).
     shell: |
-        kubectl get kubeadmcontrolplane "{{ CLUSTER_NAME }}" -n "{{ NAMESPACE }}" -o json | jq '.spec.rolloutStrategy.rollingUpdate.maxSurge=1' | kubectl apply -f-
+        kubectl get kubeadmcontrolplane "{{ CLUSTER_NAME }}" -n "{{ NAMESPACE }}" -o json | jq '.spec.rolloutStrategy.rollingUpdate.maxSurge=1' | jq '.metadata|.resourceVersion=="{{ resource_version_kcp }}"' | kubectl apply -f-
```

Comment:
Are you assigning a value ? or selecting or running conditionally ?

Reply:
Yes, I am trying to assign a value of `resourceVersion` fetched just before applying it to object.